src/arrayMethodJoin.js

```diff
@@ -4,8 +4,21 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    const strSeparator = String(separator);
+    let result = '';
+
+    for (const el of this) {
+      let strEl = el;
+
+      if (el === undefined || el === null) {
+        strEl = '';
+      }
+
+      result += strEl + strSeparator;
+    }
+
+    return result.slice(0, result.length - strSeparator.length);
```

Comment:
I don't sure can we use "slice" method in this task, but we can solve this task without "slice" method.